staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -31,147 +31,163 @@ import (
 )
 
 // NestedFieldCopy returns a deep copy of the value of a nested field.
-// false is returned if the value is missing.
-// nil, true is returned for a nil field.
-func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{}, bool) {
-	val, ok := nestedFieldNoCopy(obj, fields...)
-	if !ok {
-		return nil, false
-	}
-	return runtime.DeepCopyJSONValue(val), true
+// Returns false if the value is missing.
+// No error is returned for a nil field.
+func NestedFieldCopy(obj map[string]interface{}, fields ...string) (interface{}, bool, error) {
+	val, found, err := nestedFieldNoCopy(obj, fields...)
+	if !found || err != nil {
+		return nil, found, err
+	}
+	return runtime.DeepCopyJSONValue(val), true, nil
 }
 
-func nestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, bool) {
+func nestedFieldNoCopy(obj map[string]interface{}, fields ...string) (interface{}, bool, error) {
 	var val interface{} = obj
 	for _, field := range fields {
 		if m, ok := val.(map[string]interface{}); ok {
 			val, ok = m[field]
 			if !ok {
-				return nil, false
+				return nil, false, nil
 			}
 		} else {
-			// Expected map[string]interface{}, got something else
-			return nil, false
+			return nil, true, fmt.Errorf("%v is of the type %T, expected map[string]interface{}", val, val)
```

Comment:
tending to return false here. @ash2k @nikhita wdyt?

Reply:
I think the general convention in Go is that if an `error` is returned then all other values should be ignored. Unless documented otherwise. So it should not matter but I agree that it is better to make it `false`.

Reply:
:+1: created https://github.com/kubernetes/kubernetes/pull/56563 for this commit.